Framework/Indexing/src/IndexInfo.cpp

```diff
@@ -133,10 +123,81 @@ void IndexInfo::setSpectrumDefinitions(
   m_spectrumDefinitions = spectrumDefinitions;
 }
 
+/// Returns the spectrum definitions.
 const Kernel::cow_ptr<std::vector<SpectrumDefinition>> &
 IndexInfo::spectrumDefinitions() const {
   return m_spectrumDefinitions;
 }
 
+/** Creates an index set containing all indices.
+ *
+ * If there are multiple partitions (MPI ranks), the returned set contains the
+ * subset of indices on this partition. */
+SpectrumIndexSet IndexInfo::makeIndexSet() const {
+  makeSpectrumNumberTranslator();
+  return m_spectrumNumberTranslator->makeIndexSet();
+}
+
+/** Creates an index set containing all indices with spectrum number between
+ * `min` and `max`.
+ *
+ * If there are multiple partitions (MPI ranks), the returned set contains the
+ * subset of indices on this partition. */
+SpectrumIndexSet IndexInfo::makeIndexSet(SpectrumNumber min,
+                                         SpectrumNumber max) const {
+  makeSpectrumNumberTranslator();
+  return m_spectrumNumberTranslator->makeIndexSet(min, max);
+}
+
+/** Creates an index set containing all indices with global index between `min`
+ * and `max`.
+ *
+ * If there are multiple partitions (MPI ranks), the returned set contains the
+ * subset of indices on this partition. */
+SpectrumIndexSet IndexInfo::makeIndexSet(GlobalSpectrumIndex min,
+                                         GlobalSpectrumIndex max) const {
+  makeSpectrumNumberTranslator();
+  return m_spectrumNumberTranslator->makeIndexSet(min, max);
+}
+
+/** Creates an index set containing all indices corresponding to the spectrum
+ * numbers in the provided vector.
+ *
+ * If there are multiple partitions (MPI ranks), the returned set contains the
+ * subset of indices on this partition. */
+SpectrumIndexSet IndexInfo::makeIndexSet(
+    const std::vector<SpectrumNumber> &spectrumNumbers) const {
+  makeSpectrumNumberTranslator();
+  return m_spectrumNumberTranslator->makeIndexSet(spectrumNumbers);
+}
+
+/** Creates an index set containing all indices corresponding to the global
+ * indices in the provided vector.
+ *
+ * If there are multiple partitions (MPI ranks), the returned set contains the
+ * subset of indices on this partition. */
+SpectrumIndexSet IndexInfo::makeIndexSet(
+    const std::vector<GlobalSpectrumIndex> &globalIndices) const {
+  makeSpectrumNumberTranslator();
+  return m_spectrumNumberTranslator->makeIndexSet(globalIndices);
+}
+
+void IndexInfo::makeSpectrumNumberTranslator() const {
+  // To support legacy code that creates workspaces with duplicate spectrum
+  // numbers we are currently creating the SpectrumNumberTranslator only when
+  // needed.
+  // Note that the lazy initialization implies that no sharing of
+  // SpectrumNumberTranslator will be set up for workspaces created before the
+  // first access.
+  // TODO We are not setting monitors currently. This is ok as long as we have
+  // exactly one partition.
+  PartitionIndex partition = 0;
```

Comment:
I take it this hard-coded value will be calculated somehow at a later date. If you didn't have a follow up PR, I would have said that a comment to that effect would be needed here.